internal/protocolv6provider/server.go

```diff
@@ -117,5 +121,21 @@ func Server() tfprotov6.ProviderServer {
 		dataSourceRouter: dataSourceRouter{
 			"corner_v6_time": dataSourceTime{},
 		},
+		functions: map[string]*tfprotov6.Function{
+			"bool": {
+				Parameters: []*tfprotov6.FunctionParameter{
+					{
+						Name: "param",
```

Comment:
During the initial configuration of the `Server()` func to add functions, the `Name` field of the parameter was omitted, giving rise to the following error during execution of the `TestAccV6FunctionBool` test:

```shell
=== RUN   TestAccV6FunctionBool
    function_bool_test.go:17: Step 1/1 error: Error running pre-apply plan: exit status 1
        
        Error: Call to unknown function
        
          on terraform_plugin_test.tf line 13, in output "test":
          13: 					value = provider::corner::bool(true)
        
        There are no functions in namespace "provider::corner::".
--- FAIL: TestAccV6FunctionBool (0.55s)
```

Executing the test with `TF_LOG=TRACE` did not reveal that the issue was a consequence of the `Name` field being an empty string (refer to [log output](https://gist.github.com/bendbennett/b70429492819beefc0ffcd22485cc753)). 

This is unlikely to be an issue from the standpoint of provider developers adding provider-defined functions as _terraform-plugin-framework_ automatically adds the string "param" to function parameter `Name` fields, if the field is unpopulated, but I thought it might be worth mentioning in case the error generated by Terraform should be more explicit.

Reply:
Would you be able to double check this with the latest alpha or beta1 (potentially releasing later today)? There was some work done in core to improve various function errors, e.g. https://github.com/hashicorp/terraform/pull/34683

If it still doesn't seem reasonable, I would raise an issue in that issue tracker! 😄

Reply:
https://github.com/hashicorp/terraform/issues/34777